src/documents/tutorial-developerIntroduction/DeveloperIntroductionToInfusionFramework-EventsAndInversionOfControl.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Developer Introduction to the Infusion Framework - Events and Inversion of Control
+layout: default
+category: Tutorials
+---
+
+All Infusion components support highly flexible event-driven programming. All components have basic [lifecycle events](/infusion/development/ComponentLifecycle.md) such as creation and destruction, and others may be [declared](/infusion/development/InfusionEventSystem.md#declaring-an-event-on-a-component), [fired](/infusion/development/InfusionEventSystem.md#using-events-and-listeners-procedurally) and [listened](/infusion/development/InfusionEventSystem.md#registering-a-listener-to-an-event) for by the originating component or another component.
+
+Infusion also makes extensive use of the programming concept of [Inversion of Control](/infusion/development/FrameworkConcepts.md#ioc) (IoC), a technique for organizing component dependencies and references in a distributed, flexible manner. IoC is used in many ways throughout Infusion, but an important initial concept to grasp is the use of [IoC References](/infusion/development/IoCReferences.md) when configuring components, and the concept of `{that}`.
+
+IoC references allow us to refer to other parts of a component object (or another component entirely) in a declarative, context-specific manner, with `{that}` standing in for the current component configuration.
+
+We can use a listener definition, the `onCreate` lifecycle event and IoC `{that}` to have a component say hello when it's ready, rather than needing to be manually called:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/LxbPQZ?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.component"],
+    listeners: {
+        // Configures the component to call its own 'sayHello' function
+        // when it's created
+        //
+        // On the left side, "onCreate.sayHello":
+        // 1) Configures the listener to listen for
+        // the onCreate lifecycle event
+        // 2) Adds an a namespace of 'sayHello' to
+        // allow multiple listeners to be attached to the
+        // same event without collision
+        //
+        // On the right side, this configures the listener
+        // to use the sayHello invoker 
+        "onCreate.sayHello": "{that}.sayHello"
+    },
+    invokers: {
+        sayHello: {
+            "this": "console",
```

Comment:
Ok - another possibility is that you just write your own global function called "examples.log" that just calls console.log.

One global issue just struck me about this, which is that all your code is in the "fluid" namespace, which is not something we want to encourage in general user code - since all they are going to do is cut and paste this material and start working with it. The whole point of having the namespaces is to assist people from different communities to keep their work separate - so I suggest some different namespace for these examples such as examples.* or tutorial.*